src/extension.ts

```diff
@@ -149,6 +149,8 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
                 node = await ext.tree.showTreeItemPicker<MongoDocumentTreeItem | DocDBDocumentTreeItem>([MongoDocumentTreeItem.contextValue, DocDBDocumentTreeItem.contextValue], actionContext);
             }
 
+            // Clear un-uploaded local changes to the document before opening https://github.com/microsoft/vscode-cosmosdb/issues/1619
+            ext.fileSystem.fireChangedEvent(node);
```

Comment:
Actually any chance you could do this in [here](https://github.com/microsoft/vscode-azuretools/blob/master/ui/src/AzExtTreeFileSystem.ts#L40)? If you don't want to do it for this release candidate that's fine, but I think other extensions will have the same problem, so I want it in the shared package eventually

Reply:
Sounds good. I logged a new issue for it: https://github.com/microsoft/vscode-cosmosdb/issues/1624

Reply:
Okay well this one will take longer to discuss during triage than it'll take to fix. So if you don't want to do it within like a day - I'll just do it myself haha. Let me know

Reply:
Lol gotcha, I can take care of it right after we ship tomorrow